tests/jointcalTestBase.py

```diff
@@ -0,0 +1,311 @@
+# See COPYRIGHT file at the top of the source tree.
+
+from __future__ import division, absolute_import, print_function
+
+import numpy as np
+import collections
+import os
+
+from lsst.meas.astrom import LoadAstrometryNetObjectsTask, LoadAstrometryNetObjectsConfig
+from lsst.afw import geom, table
+
+from lsst.jointcal import jointcal
+
+arcsec_per_radian = 206265.
+
+
+class JointcalTestBase(object):
+    """
+    Base class for jointcal tests, to genericize some test running and setup.
+
+    Derive from this first, then from TestCase.
+    """
+
+    def setUp(self):
+        self.doPlot = False  # don't make plots unless specifically requested
+        self.matchRadius = 0.1*geom.arcseconds  # match sources within 0.1" for RMS statistics
+
+    def tearDown(self):
+        if getattr(self, 'jointcalTask', None) is not None:
+            del self.jointcalTask
+        if getattr(self, 'jointcalTask', None) is not None:
+            del self.catalogs
+
+    def _prep_reference_loader(self, center, radius):
+        """
+        Setup an astrometry.net reference loader.
+
+        @param center (afw.coord) The center of the field you're testing on.
+        @param radius (afw.geom.angle) The radius to load objects around center.
+        """
+        refLoader = LoadAstrometryNetObjectsTask(LoadAstrometryNetObjectsConfig())
+        # Make a copy of the reference catalog for in-memory contiguity.
+        self.reference = refLoader.loadSkyCircle(center, radius, filterName='r').refCat.copy()
+
+    def _testJointCalTask_run(self, nCatalogs, relative_error, absolute_error):
+        """Test jointcal.run() on nCatalogs, requiring less than some error (arcsec)."""
+
+        result = self.jointcalTask.run(self.catalogs[:nCatalogs])
+        self.dataRefs = result.dataRefs
+        self.old_wcss = result.old_wcss
+
+        rms_rel, rms_abs = self.compute_rms(self.catalogs[:nCatalogs], fluxlimit=100)
+        self.assertLess(rms_rel, relative_error/arcsec_per_radian)
+        self.assertLess(rms_abs, absolute_error)
+
+    def _testJointCalTask(self, nCatalogs, relative_error, absolute_error):
+        """Test parseAndRun for jointcal on nCatalogs, requiring less than some error (arcsec)."""
+
+        visits = '^'.join(str(v) for v in self.visitList[:nCatalogs])
+        output_dir = os.path.join('.test', self.__class__.__name__)
+        result = jointcal.JointcalTask.parseAndRun(args=[self.input_dir, '--output', output_dir,
+                                                         '--clobber-versions', '--clobber-config',
+                                                         '--doraise',
+                                                         '--id', 'visit=%s'%visits],
```

Comment:
Well, you've got me there.  I think you're the first person to actually do what our coding standard says in this respect...